### PR TITLE
Add token refresh to Synapse for writing to MPI

### PR DESCRIPTION
**Pull request by m-goggins**
- Adds a token refresh function to update the bearer token to hit the ingestion and record linkage endpoints for more than 1 hour should there be a large MPI file, e.g., during seeding

### CODE REVIEW

File: scripts/Synapse/convertParquetMPI.ipynb [hunk 1]
```diff
@@ -25,8 +25,8 @@
       "metadata": {},
       "outputs": [],
       "source": [
-        "### TEMP WORKAROUND CELL\n",
-        "# pip install phdi==1.0.6\n",
+        "## TEMP WORKAROUND\n",
+        "# pip install phdi\n",
```
Comment (bamader) [inline scripts/Synapse/convertParquetMPI.ipynb:29]: Do we need this since we install above?

File: scripts/Synapse/convertParquetMPI.ipynb [hunk 2]
```diff
@@ -186,8 +196,17 @@
...
-        "ingestion_data, ingestion_headers = set_up_endpoint_access(endpoint=\"ingestion\",url=url)\n",
+        "record_linkage_data, record_linkage_headers = set_up_endpoint_access(endpoint=\"record-linkage\",url=endpoint_login_url)\n",
+        "ingestion_data, ingestion_headers = set_up_endpoint_access(endpoint=\"ingestion\",url=endpoint_login_url)\n",
+        "\n",
+        "# Define function for refreshing Bearer access token\n",
+        "def request_with_token_refresh(url,headers,data,endpoint_type,endpoint_login_url):\n",
+        "    resp = http_request_with_retry(url=url,retry_count=2,request_type=\"POST\",headers=headers,allowed_methods=[\"POST\"],data=data)\n",
+        "    if resp.status_code == 401:\n",
+        "        _, headers = set_up_endpoint_access(endpoint=endpoint_type,url=endpoint_login_url)\n",
+        "        return request_with_token_refresh(url,headers,data,endpoint_type,endpoint_login_url)\n",
```
Comment (bamader) [inline scripts/Synapse/convertParquetMPI.ipynb:207]: Nothing better than seeing a recursive function out in the wild 😎

**Review by bamader — APPROVED** [review]
Looks good! Just the one comment around the phdi installation.

File: scripts/Synapse/convertParquetMPI.ipynb (hunk 1 above)
Comment (m-goggins) [reply scripts/Synapse/convertParquetMPI.ipynb:29]: Good catch. we do not!

**Pull request merged**